packages/ckb/src/rgbpp/btc-transfer.ts

```diff
@@ -163,3 +187,186 @@ export const genBtcTransferCkbVirtualTx = async ({
     sumInputsCapacity: append0x(sumInputsCapacity.toString(16)),
   };
 };
+
+/**
+ * Generate the virtual ckb transaction for the btc batch transfer tx
+ * @param collector The collector that collects CKB live cells and transactions
+ * @param xudtTypeBytes The serialized hex string of the XUDT type script
+ * @param rgbppLockArgsList The rgbpp assets cell lock script args array whose data structure is: out_index | bitcoin_tx_id
+ * @param rgbppReceivers The rgbpp receiver list which include toBtcAddress and transferAmount
+ * @param isMainnet
+ */
+export const genBtcBatchTransferCkbVirtualTx = async ({
+  collector,
+  xudtTypeBytes,
+  rgbppLockArgsList,
+  rgbppReceivers,
+  isMainnet,
+}: BtcBatchTransferVirtualTxParams): Promise<BtcBatchTransferVirtualTxResult> => {
+  const xudtType = blockchain.Script.unpack(xudtTypeBytes) as CKBComponents.Script;
+
+  if (!isTypeAssetSupported(xudtType, isMainnet)) {
+    throw new TypeAssetNotSupportedError('The type script asset is not supported now');
+  }
+
+  const rgbppLocks = rgbppLockArgsList.map((args) => genRgbppLockScript(args, isMainnet));
+  let rgbppCells: IndexerCell[] = [];
+  for await (const rgbppLock of rgbppLocks) {
+    const cells = await collector.getCells({ lock: rgbppLock, type: xudtType });
+    if (!cells || cells.length === 0) {
+      throw new NoRgbppLiveCellError('No rgbpp cells found with the xudt type script and the rgbpp lock args');
+    }
+    rgbppCells = [...rgbppCells, ...cells];
+  }
+  rgbppCells = rgbppCells.sort(compareInputs);
+
+  const sumTransferAmount = rgbppReceivers
+    .map((receiver) => receiver.transferAmount)
+    .reduce((prev, current) => prev + current, BigInt(0));
+
+  const rpbppCellCapacity = calculateRgbppCellCapacity(xudtType);
+  const outputs: CKBComponents.CellOutput[] = rgbppReceivers.map((_, index) => ({
+    // The Vouts[0] for OP_RETURN and Vouts[1], Vouts[2], ... for RGBPP assets
+    lock: genRgbppLockScript(buildPreLockArgs(index + 1), isMainnet),
+    type: xudtType,
+    capacity: append0x(rpbppCellCapacity.toString(16)),
+  }));
+  const outputsData = rgbppReceivers.map((receiver) => append0x(u128ToLe(receiver.transferAmount)));
+
+  const { inputs, sumInputsCapacity, sumAmount } = collector.collectUdtInputs({
+    liveCells: rgbppCells,
+    needAmount: sumTransferAmount,
+    isMax: true,
```

Comment:
### suggestion:
- https://github.com/ckb-cell/rgbpp-sdk/issues/116

Reply:
I think the parameter `rgbppCells` has already specified that this is RGB++ lock, so is it redundant to check again?

Reply:
> I think the parameter `rgbppCells` has already specified that this is RGB++ lock, so is it redundant to check again?

Yes.

So it sounds like the `isMax` param is not required.
Because it should be true, when the inputs are `rgbppCells`.

Reply:
OK. I see. Make sense